ovirt_imageio/_internal/backends/http.py

```diff
@@ -613,6 +620,12 @@ def server_address(self):
         return self.sock.getpeername()[:2]
 
 
+class HTTPSConnection(HTTPConnection, http_client.HTTPSConnection):
+    """
+    Enhanced HTTPS connection.
+    """
```

Comment:
Not sure this works correctly - HTTPConnection methods may override HTTPSConnection methods in an unwanted ways. But even if it works this make the code much more
complex and hard to understand and modify.

It will be simpler and safer to use a mixin for adding is_local and override the server_address:

     class ConnectionMixin:
         def is_local(...)
         def server_address(...)

     class HTTPConnection(ConnectionMixin, http_client.HTTPConnection):
         pass

     class HTTPSConnection(ConnectionMixin, http_client.HTTPSConnection):
         pass

Reply:
I am pretty sure it works as intended, but I had to specifically look for multiple inheritance and method override to be sure it'd work. Which is a good indicative that it might be a bit complex.

Take into account that the `is_local` and `server_address` access member variables and methods that are not defined unless we inherit from `http_client.HTTPConnection`, which will result in a code similar to what I have done:
```python
 class ConnectionMixin(http_client.HTTPConnection):
     def is_local(...)
     def server_address(...)

 class HTTPConnection(ConnectionMixin):
     pass

 class HTTPSConnection(ConnectionMixin, http_client.HTTPSConnection):
     pass

```